trending_data/src/main/java/com/google/sps/data/Country.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+
+public class Country {
+  String name = "";
+  String alpha2Code = "";
+  double lat;
+  double lng;
+
+  // Given a String in this format:
+  // "Country","Alpha-2 code","Alpha-3 code","Numeric code","Latitude (average)","Longitude
+  // (average)"
+  // , creates the Country object which contains country name , alpha-2 code and the  lat,lng
+  public Country(String s) {
+    // true if the index (i) is inside a pair of ("")
+    // will fail if string has a (") inside a pair of ("")
+    ArrayList<String> fields = new ArrayList();
+    boolean insideQuote = false;
+    StringBuilder token = new StringBuilder();
+    for (int i = 0; i < s.length(); i++) {
```

Comment:
You could put this functionality inside a method -> e.g. `public static List<String> parse(String s)`

There's also some existing java functions for strings that might make the code more readable. You could for example do:
```
List<String> fields = Array.asList(s.split(",")); // to split comma separated list

for (field : fields) {
  field = field.replace("\"", ""); // eliminate the double quotations
}

return fields;
```

Reply:
I don't know if having a method will make it more readable because that's all the constructor does anyway so It would be just 
`public Country() {
  parse();
  then just assign the variables here
}`
especially that I won't be parsing anywhere else and then the method would also be private so it would help in any other way.

I didn't use replace because some country names have commas (,) inside them so it would break in these cases.But there is no country name that has a (") in it. I was actually thinking that I can throw an exception or let java throw an exception if there is something wrong (basically remove the if so that if anything goes wrong we know that some country name isn't working)

Reply:
I don't see any country name with a (,) in it. Could you give a specific example?
The `replace` removes double quotations after separating the string into fields.

so 1st step is: `" "Albania", "AL", "ALB", "8", "41", "20" "` -> `["Albania", "AL", "ALB", "8", "41", "20"]`
then: `["Albania", "AL", "ALB", "8", "41", "20"]` -> `[Albania, AL, ALB, 8, 41, 20]`

Reply:
I was thinking
```
public Country(String s) {
  List<String> fields = parse(s);

  // then initialise fields
}
```